### PR TITLE
(#4898) - allow creating databases with reserved words

### PR DESCRIPTION
**Pull request by nolanlawson**
Just cleaning house, noticed that we were using regular objects here when we should be using `Map`s. Some other small fixes inside, although nothing earth-shattering.

### CODE REVIEW

File: src/adapter.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { extend as extend } from 'js-extend';
 import Promise from './deps/promise';
 import pick from './deps/pick';
-import collections from 'pouchdb-collections';
+import { Map } from 'pouchdb-collections';
```
Comment (nolanlawson) [inline src/adapter.js]: Using this style is clearer (IMO) and leads to slightly more minified code.

File: src/adapters/idb/index.js [hunk 2]
```diff
@@ -790,17 +791,16 @@ function init(api, opts, callback) {
     idbChanges.removeAllListeners(dbName);
 
     //Close open request for "dbName" database to fix ie delay.
-    if (IdbPouch.openReqList[dbName] && IdbPouch.openReqList[dbName].result) {
```
Comment (nolanlawson) [inline src/adapters/idb/index.js]: There was no reason to export the `openReqList` object.

File: tests/integration/test.basics.js [hunk 3]
```diff
@@ -697,6 +697,13 @@ adapters.forEach(function (adapter) {
       });
     });
 
+    it('Create a db with a reserved name', function () {
+      var db = new PouchDB('__proto__');
+      return db.info().then(function () {
+        return db.destroy();
+      });
+    });
+
```
Comment (nolanlawson) [inline tests/integration/test.basics.js]: The one test I added. Hopefully it won't push us over the Safari limit.

File: src/constructor.js [hunk 4]
```diff
@@ -29,8 +29,6 @@ function prepareForDestruction(self, opts) {
 
   function onDestroyed() {
     ctor.emit('destroyed', name);
-    //so we don't have to sift through all dbnames
-    ctor.emit(name, 'destroyed');
```
Comment (nolanlawson) [inline src/constructor.js]: This is an untested and undocumented feature that breaks if you try to emit a string like `'__proto__'`. I'm not aware of any code that is using it either in core or in plugin-land, so I propose we just remove it.

File: src/adapter.js (hunk 1 above)
Comment (nickcolley) [inline src/adapter.js]: :+1:

**Comment by daleharvey** [comment]
Looks good to me and all passing - https://github.com/pouchdb/pouchdb/commit/14449b7f80d27c6c08ec6c180dd8b5c1eb1439c0 :+1:

**Pull request closed without merging**